controllers/storagecluster/consumer.go

```diff
@@ -0,0 +1,177 @@
+package storagecluster
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	providerClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+)
+
+var (
+	externalOCSResources map[types.UID][]ExternalResource
+)
+
+type ocsConsumer struct{}
+
+func (o *ocsConsumer) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	if !(instance.Spec.ExternalStorage.Enable && instance.Spec.ExternalStorage.StorageProviderKind == ocsv1.KindOCS) {
+		return nil
+	}
+
+	if instance.Status.ExternalStorage.ConsumerID == "" {
+		return o.onBoardConsumer(r, instance)
+	}
+
+	if instance.Status.ExternalStorage.GrantedCapacity.Cmp(*instance.Spec.ExternalStorage.RequestedCapacity) == -1 {
```

Comment:
What is the requested capacity is less? (that is the user wan't to downscale)

Reply:
changed the code